dbms/src/TestUtils/MPPTaskTestUtils.cpp

```diff
@@ -0,0 +1,128 @@
+{
+    ExecutorTest::SetUpTestCase();
+    log_ptr = Logger::get("compute_test");
+    server_num = 1;
+}
+
+void MPPTaskTestUtils::TearDown()
+{
+    MockComputeServerManager::instance().reset();
+}
+
+void MPPTaskTestUtils::startServers()
+{
+    startServers(server_num);
+}
+
+void MPPTaskTestUtils::startServers(size_t server_num_)
+{
+    server_num = server_num_;
+    test_meta.server_num = server_num_;
+    test_meta.context_idx = TiFlashTestEnv::globalContextSize();
+    MockComputeServerManager::instance().reset();
+    auto size = std::thread::hardware_concurrency();
+    GRPCCompletionQueuePool::global_instance = std::make_unique<GRPCCompletionQueuePool>(size);
+    for (size_t i = 0; i < test_meta.server_num; ++i)
+    {
+        MockComputeServerManager::instance().addServer(MockServerAddrGenerator::instance().nextAddr());
+        // Currently, we simply add a context and don't care about destruct it.
+        TiFlashTestEnv::addGlobalContext();
+        TiFlashTestEnv::getGlobalContext(i + test_meta.context_idx).setMPPTest();
+    }
+
+    MockComputeServerManager::instance().startServers(log_ptr, test_meta.context_idx);
+    MockServerAddrGenerator::instance().reset();
+}
+
+size_t MPPTaskTestUtils::serverNum()
+{
+    return server_num;
+}
+
+std::tuple<size_t, std::vector<BlockInputStreamPtr>> MPPTaskTestUtils::prepareMPPStreams(DAGRequestBuilder builder)
+{
+    auto properties = DB::tests::getDAGPropertiesForTest(serverNum());
+    auto tasks = builder.buildMPPTasks(context, properties);
+    for (int i = test_meta.context_idx; i < TiFlashTestEnv::globalContextSize(); ++i)
+        TiFlashTestEnv::getGlobalContext(i).setCancelTest();
+    MockComputeServerManager::instance().setMockStorage(context.mockStorage());
+    auto res = executeMPPQueryWithMultipleContext(properties, tasks, MockComputeServerManager::instance().getServerConfigMap());
+    return {properties.start_ts, res};
+}
+
+ColumnsWithTypeAndName MPPTaskTestUtils::exeucteMPPTasks(QueryTasks & tasks, const DAGProperties & properties, std::unordered_map<size_t, MockServerConfig> & server_config_map)
+{
+    auto res = executeMPPQueryWithMultipleContext(properties, tasks, server_config_map);
+    return readBlocks(res);
+}
+
+String MPPTaskTestUtils::queryInfo(size_t server_id)
+{
+    FmtBuffer buf;
+    buf.fmtAppend("server id: {}, tasks: ", server_id);
+    buf.fmtAppend(TiFlashTestEnv::getGlobalContext(server_id).getTMTContext().getMPPTaskManager()->toString());
+    return buf.toString();
+}
+
+::testing::AssertionResult MPPTaskTestUtils::assertQueryCancelled(size_t start_ts)
+{
+    auto seconds = std::chrono::seconds(1);
+    auto retry_times = 0;
+    for (int i = test_meta.context_idx; i < TiFlashTestEnv::globalContextSize(); ++i)
+    {
+        // wait until the task is empty for <query:start_ts>
+        while (TiFlashTestEnv::getGlobalContext(i).getTMTContext().getMPPTaskManager()->getQueryTaskSetWithoutLock(start_ts) != nullptr)
```

Comment:
why don't need lock here?

Reply:
> why don't need lock here?

Because I only want to check the status of the Task and will not change the status.